src/components/FullHierarchyTable/FullHierarchyTable.tsx

```diff
@@ -202,38 +395,45 @@ export const FullHierarchyTable: React.FunctionComponent<FullHierarchyTableAllPr
                     }
                 }))
         ]
-    }, [indentLevel, fields, props.meta.name])
-
-    return <div className={styles.container}>
-        <Table
-            className={styles.table}
-            rowSelection={rowSelection}
-            rowKey="id"
-            columns={columns}
-            pagination={false}
-            showHeader={depthLevel === 1}
-            expandIcon={Exp as any}
-            defaultExpandedRowKeys={undefined}
-            expandedRowKeys={userOpenedRecords}
-            onExpand={handleExpand}
-            dataSource={tableRecords}
-            expandedRowRender={nestedHierarchy}
-            expandIconAsCell={false}
-            expandIconColumnIndex={(props.selectable) ? 1 : 0}
-            loading={props.loading}
-            onRow={!(hierarchyDisableRoot && depthLevel === 1) && props.onRow}
-        />
-    </div>
+    }, [inputPlaceholder, indentLevel, fields, props.meta.name, props.bcFilter])
+
+    return loading
+        ? <Skeleton loading paragraph={{rows: 5}}/>
+        : <div className={styles.container}>
+            <Table
+                className={styles.table}
+                rowSelection={rowSelection}
+                rowKey="id"
+                columns={columns}
+                pagination={false}
+                showHeader={depthLevel === 1}
+                expandIcon={Exp as any}
+                defaultExpandedRowKeys={undefined}
+                expandedRowKeys={userOpenedRecords || []}
+                onExpand={handleExpand}
+                dataSource={tableRecords}
+                expandedRowRender={nestedHierarchy}
+                expandIconAsCell={false}
+                expandIconColumnIndex={(props.selectable) ? 1 : 0}
+                loading={loading}
+                onRow={!(hierarchyDisableRoot && depthLevel === 1) && props.onRow}
+                getPopupContainer={trigger => trigger}
+            />
+        </div>
 }
 
 function mapStateToProps(store: Store, ownProps: FullHierarchyTableOwnProps): FullHierarchyTableProps {
     const bcName = ownProps.meta.bcName
     const bc = store.screen.bo.bc[bcName]
-    const loading = bc?.loading
+    const bcUrl = buildBcUrl(bcName, true)
+    const rowMeta = bcUrl && store.view.rowMeta[bcName]?.[bcUrl]
+    const loading = bc?.loading || !rowMeta
     return {
         loading: loading,
         data: (loading) ? emptyData : store.data[bcName] as AssociatedItem[],
-        pendingChanges: store.view.pendingDataChanges[bcName]
+        pendingChanges: store.view.pendingDataChanges[bcName],
+        bcFilter: store.screen.filters[bcName],
```

Comment:
The name is misleadig as it's an array, should be `filters`

Reply:
fixed `bcFilter` > `bcFilters`